polkadot/zombienet_tests/elastic_scaling/0002-elastic-scaling-doesnt-break-parachains.toml

```diff
@@ -8,6 +8,7 @@ bootnode = true
 [relaychain.genesis.runtimeGenesis.patch.configuration.config.scheduler_params]
   max_validators_per_core = 2
   num_cores = 2
+  lookahead = 3
```

Comment:
Advertisements got ignored because they were over the limit and the test failed.
Do you think it should also work with 2?

Reply:
Not sure. I'm trying to understand whether this change will break parachains on polkadot unless we also bump lookahead there

Reply:
> Advertisements got ignored because they were over the limit and the test failed. Do you think it should also work with 2?

I'm checking this further.

Reply:
Indeed there was an issue.
I'm using [`known_allowed_relay_parents_under`](https://github.com/paritytech/polkadot-sdk/blob/0b52a2c19ebcf5d7a0d07974b70aec656704d249/polkadot/node/subsystem-util/src/backing_implicit_view.rs#L315) to get all relay parents within the current view and calculate the number of pending and seconded candidates at them. However this function returns the relay parent in question + `lookahead` number of ancestors. This is a problem because the expected number of advertisements is the entries in the claim queue (which has got length equal to `lookahead`). So when checking if the seconding limit is reach I was looking at one extra relay parent.

There are two solutions for this problem:
1. Trim the blocks returned from `known_allowed_relay_parents_under` [here](https://github.com/paritytech/polkadot-sdk/blob/cb69361d0ad009eed288aa256c8f7699ae9656b1/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L2081) so that it equals `lookahead`.
2. Add the `ParaId` currently scheduled on the core to the assignments.

After a discussion with @eskimor we decided to go with 2 since it relaxes the requirements for older candidates instead of making them stricter. In other words 2 gives better chances of delayed advertisements to be included.

This complicates the code in `assign_incoming` a little bit ([this part](https://github.com/paritytech/polkadot-sdk/blob/cb69361d0ad009eed288aa256c8f7699ae9656b1/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L472-L517)) but after we deal with https://github.com/paritytech/polkadot-sdk/issues/5079 it can be simplified again.


 from When calculating the seconded and pending candidates in a view [here](https://github.com/paritytech/polkadot-sdk/blob/cb69361d0ad009eed288aa256c8f7699ae9656b1/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L2081)

Reply:
Thinking more about this issue made me change my mind. Going with 2 means we add artificial duplicated data because `availability_cores` runtime api already uses the claim queue to fill in the core data.
It makes more sense to go with 1 and when counting the occupied slots to go back as far as the length of the claim queue allows instead of the whole view (the difference always will be 1). This approach doesn't limit the oldest relay parent at which we will accept a candidate. It only fixes occupied slots counting so I believe it will work fine.

@eskimor do you agree with this?

Reply:
Sorry I don't follow why we bumped the lookahead to 4 in the test

Reply:
I've reverted the change after I found the bug mentioned in the previous comment. `lookahead` is not bumped for this test anymore.